python/dolfinx/fem/petsc.py

```diff
@@ -864,7 +864,7 @@ def __init__(
         else:
             self._u = u
 
-        self.bcs = bcs
+        self.bcs = bcs if bcs is not None else []
```

Comment:
For my OCD maybe make it `bcs = [] if bcs is None else bcs`, which is what we used in `NonlinearProblem` :)